CMakeLists.txt

```diff
@@ -172,6 +172,9 @@ endif ()
 # Enable unit tests.
 option(OPENASSETIO_ENABLE_TESTS "Create test targets" OFF)
 
+# Enable clang-format formatting check.
+option(OPENASSETIO_ENABLE_CLANG_FORMAT "Enable clang-format check during build" OFF)
+
 # Enable clang-tidy static analysis tool.
 option(OPENASSETIO_ENABLE_CLANG_TIDY "Enable clang-tidy analysis during build" OFF)
```

Comment:
[question] If these are off by default, should we have a 'dev' CMake preset that turns all this on, along with tests? Or just turn them on? It'd be great if the default experience was overzealous, as we should never be merging anything to a release branch where these two checks being `ON` would fail.

Reply:
The default really needs to be what we would expect a parent project to want if OpenAssetIO was used as a subproject.

Otherwise yeah I agree it's about time we had some presets. Is it worth doing as part of this PR or another issue?

Reply:
Ahh yes, perfect, `OFF` + dev preset sounds ideal. Thanks.

Reply:
Added #315